src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2627,14 +2627,23 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
                 // to call.
 
                 MethodDesc directMethod = constrainedType.GetClosestDefType().TryResolveConstraintMethodApprox(exactType, method, out forceUseRuntimeLookup);
-#if !READYTORUN
                 if (directMethod == null && constrainedType.IsEnum)
                 {
+#if READYTORUN
+                    if (method.Name == "GetHashCode")
```

Comment:
This should do a fast check `method.IsIntrinsic` to avoid getting the name for all methods that come through here.

Reply:
Hmm, just tried this but it doesn't seem to work, "method.IsIntrinsic" returns false for GetHashCode. Am I missing something or does this indicate some deeper inconsistency in the way we're loading assemblies or something?

Reply:
Yep, the method should be marked with the `Intrinsic` attribute in CoreCLR to make this work.

Reply:
At the very least, we're only dealing with methods on enum types, I doubt there are typically many - does today Roslyn even let you define custom methods on enums?

Reply:
The CoreRT convention is to mark all methods that are treated in special way by the codegen with Intrinsic attribute. We do this for all methods both to document the fact that the method is special cased by the codegen and to make the intrinsic identification performant.

Reply:
Makes sense, will do.